server/komodo-rest/src/main/java/org/komodo/rest/service/KomodoConnectionService.java

```diff
@@ -835,6 +849,204 @@ public Response validateConnectionName( final @Context HttpHeaders headers,
+
+            String title = RelationalMessages.getString(RelationalMessages.Info.VDB_DEPLOYMENT_STATUS_TITLE);
+            KomodoStatusObject status = new KomodoStatusObject(title);
+
+            List<String> progressMessages = deployStatus.getProgressMessages();
+            for (int i = 0; i < progressMessages.size(); ++i) {
+                status.addAttribute("ProgressMessage" + (i + 1), progressMessages.get(i));
+            }
+
+            if (deployStatus.ok()) {
+                status.addAttribute("deploymentSuccess", Boolean.TRUE.toString());
+                status.addAttribute(vdb.getName(uow),
+                                    RelationalMessages.getString(RelationalMessages.Info.VDB_SUCCESSFULLY_DEPLOYED));
+            } else {
+                status.addAttribute("deploymentSuccess", Boolean.FALSE.toString());
+                List<String> errorMessages = deployStatus.getErrorMessages();
+                for (int i = 0; i < errorMessages.size(); ++i) {
+                    status.addAttribute("ErrorMessage" + (i + 1), errorMessages.get(i));
+                }
+
+                status.addAttribute(vdb.getName(uow),
+                                    RelationalMessages.getString(RelationalMessages.Info.VDB_DEPLOYED_WITH_ERRORS));
+            }
+
+           return commit(uow, mediaTypes, status);
+        } catch ( final Exception e ) {
+            if ( ( uow != null ) && ( uow.getState() != State.ROLLED_BACK ) ) {
+                uow.rollback();
+            }
+
+            if ( e instanceof KomodoRestException ) {
+                throw ( KomodoRestException )e;
+            }
+
+            return createErrorResponseWithForbidden( headers.getAcceptableMediaTypes(), 
+                                                     e, 
+                                                     CONNECTION_SERVICE_DEPLOY_CONNECTION_VDB_ERROR );
+        }
+    }
+    
+    /**
+     * Gets the VDBStatus for all connections in the workspace.  If no VDB found for a connection,
+     * no status is returned.
+     * @param headers
+     *        the request headers (never <code>null</code>)
+     * @param uriInfo
+     *        the request URI information (never <code>null</code>)
+     * @return a JSON document the name of all connections and their VDB status in the local server (never <code>null</code>)
+     * @throws KomodoRestException
+     *         if there is a problem constructing the VDBs JSON document
+     */
+    @GET
+    @Path(V1Constants.VDB_STATUS_SEGMENT)
+    @Produces( MediaType.APPLICATION_JSON )
+    @ApiOperation(value = "Display the VDBStatus of all connections in the workspace",
+                  response = RestObjectVdbStatus.class)
+    @ApiResponses(value = {
+        @ApiResponse(code = 403, message = "An error has occurred.")
+    })
+    public Response getVdbStatuses(final @Context HttpHeaders headers,
+                                   final @Context UriInfo uriInfo) throws KomodoRestException {
+
```

Comment:
This should be moved to KomodoMetadataService and made generic for any named artifact, ie. connection or dataservice.

This ends up being really confusing for the API. We would have:
* a vdb resource that provides GET information of vdbs
* a connection resource that provides GET information on connections but also now provides GET information on connection vdbs too

Do we really want to expose the vehicle that a connection needs to be deployed in, ie. a vdb, as it is a rather confusing implementation-only concept?

This maybe should be getDeployedConnectionStatuses (although something more pithy and clear would be better).